src/index.ts

```diff
@@ -60,9 +63,41 @@ export async function downloadEntityAndContentFiles(
   }
 
   const entityMetadata: {
+    type: string,
+    metadata?: any,
     content?: Array<ContentMapping>
   } = JSON.parse(contentStream)
 
+  if (entityMetadata.type === 'profile' && entityMetadata.metadata) {
+    /*
+     * Profiles can have some images referenced in the avatar snapshots that are not included in content section.
+     * Why can this happen? Because a previous version of the profile did include those images in the content, and
+     * later on a new version of the profile decided not to include it (perhaps to avoid uploading it again) but is
+     * still referencing it in snapshots.
+     * This fix downloads those files not referenced in content but that are anyway referenced from snapshots.
+     * A proper fix needs to be added that validates and forces new deployments to include the files in content (even
+     *  if no need to upload them again).
+     */
+    const allAvatars: any[] = entityMetadata.metadata?.avatars ?? []
+    const snapshots = allAvatars.flatMap(avatar => Object.values(avatar.avatar.snapshots ?? {}) as string[])
+        .filter(snapshot => !!snapshot)
+        .filter(snapshot => !entityMetadata.content || entityMetadata.content.find(content => content.hash === snapshot) === undefined)
+    if (snapshots.length > 0) {
+      logger.info(`Downloading snapshots ${snapshots} for fixing entity ${JSON.stringify(entityMetadata)}`)
+      await Promise.all(
+          snapshots.map(snapshot => downloadFileWithRetries(
```

Comment:
how does this work? It takes only the values ignoring the keys?

Reply:
Yes. The map looks like this:
```
          "snapshots": {
            "face": "QmWxyDrJWABXjGonFpUxJD8YYzz2xiFNxupGwYKTbySaZD",
            "face128": "QmUiEzCQPxz5eHq7KXMrGq7PiM1fnZNvZg2sWELQnuYank",
            "face256": "Qma14gteWwHrn61kv4zkRABKzopWd5ZXBppHRhbqikfaev",
            "body": "QmazJLZfUmZgNMTdwWSmJRvw4dBfcjS9GuqkwkKGRWb4K6"
          },
```

Reply:
We take those ids (the values) and download the entities only if they are not included in content array.